dev/io.openliberty.org.jboss.resteasy.common.ee10/src/io/openliberty/org/jboss/resteasy/common/client/LibertyResteasyClientBuilderImpl.java

```diff
@@ -81,6 +80,18 @@ public ResteasyClient build() {
            }
         }
 
+        if (asyncExecutor == null) {
+            cleanupExecutor = false;
+            OsgiFacade facade = OsgiFacade.instance().orElse(null);
+            if (facade != null) {
+                Object serviceRef = facade.getServiceRef(ExecutorService.class).orElse(null);
```

Comment:
Is this way of obtaining the Liberty global executor sufficiently non-brittle?  Code elsewhere includes a filter, such as `(component.name=com.ibm.ws.threading)` to ensure the correct one is obtained.

Reply:
I tried to find an example of what you are referring to.  I couldn't find one.  Can you point me to an example?

Reply:
Sure, here are some examples,
https://github.com/OpenLiberty/open-liberty/blob/adec923541de1b7cfa32414304f62f867feea113/dev/com.ibm.jbatch.container/src/com/ibm/jbatch/container/services/impl/JPAPersistenceManagerImpl.java#L194
https://github.com/OpenLiberty/open-liberty/blob/adec923541de1b7cfa32414304f62f867feea113/dev/com.ibm.ws.concurrent.persistent/src/com/ibm/ws/concurrent/persistent/internal/PersistentExecutorImpl.java#L1839
https://github.com/OpenLiberty/open-liberty/blob/adec923541de1b7cfa32414304f62f867feea113/dev/com.ibm.ws.threading/src/com/ibm/ws/threading/CompletionStageFactory.java#L47
https://github.com/OpenLiberty/open-liberty/blob/adec923541de1b7cfa32414304f62f867feea113/dev/com.ibm.ws.threading/src/com/ibm/ws/threading/PolicyExecutorProvider.java#L47

But it looks like there are even more examples to the contrary, so it looks like no one worries about this.

Reply:
After doing some testing I ended up doing a different way to resolve this.  As such I was able to do what you were asking for.  Can you take another look at the changes?